cpp/src/fil/common.cuh

```diff
@@ -173,6 +172,89 @@ struct predict_params : shmem_size_params {
   int num_blocks;
 };
 
+namespace dispatch {
+
+template <template <bool, leaf_algo_t, int> class Func, typename storage_type,
+          bool cols_in_shmem, leaf_algo_t leaf_algo, int n_items,
```

Comment:
Consider using `N_ITEMS` for the template parameter, to make it more obvious where it is used.